src/input/keyboard.c

```diff
@@ -592,11 +592,13 @@ keyboard_update_layout(struct seat *seat, xkb_layout_index_t layout)
 		kb->modifiers.latched, kb->modifiers.locked, layout);
 }
 
-void
-keyboard_init(struct seat *seat)
+/*
+ * Set layout based on environment variables XKB_DEFAULT_LAYOUT,
+ * XKB_DEFAULT_OPTIONS, and friends.
+ */
+static void
+set_layout(struct wlr_keyboard *kb)
```

Comment:
I think we should prevent setting the same keymap again. We can test for that with `wlr_keyboard_keymaps_match()`. 

That would also allow to call `reset_window_keyboard_layout_groups()` only when changing the keymap without us requiring to compare the single env var. The user might also have changed some system layout files or used other environment vars that influence the layout.

Reply:
Nice catch. I didn't know about `wlr_keyboard_keymaps_match()`

Reply:
> That would also allow to call reset_window_keyboard_layout_groups() only when changing the keymap without us being required to compare the single env var. The user might also have changed some system layout files or used other environment vars that influence the keymap.

I'm not sure that works. For example, what if a user has changed group (e.g. `de` -> `se`) before a `--reconfigure`. That would then trigger a `reset_window_keyboard_layout_groups()` even though the `XKB_DEFAULT_LAYOUT` didn't change. Or is my brain just not thinking straight?

Reply:
> For example, what if a user has changed group (e.g. `de` -> `se`) before a `--reconfigure`.

Not sure I follow. You mean via `XKB_DEFAULT_LAYOUT`?

As long as the user doesn't change any system xkb files or one of the `XKB_` vars before a `--reconfigure` the keymap should be the same and thus we would not call `reset_window_keyboard_layout_groups()`.

Reply:
Yes, I'm stupid. I was thinking `group` rather than `keymap`. Have pushed fixup. Works fine now.